core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -147,8 +143,8 @@ public void stop() {
             loadingAds.cancel();
             loadingAds = null;
         }
-        if (adPlaybackState != null && player != null && player.isPlayingAd()) {
-            adPlaybackState = adPlaybackState.withAdResumePositionUs(TimeUnit.MILLISECONDS.toMicros(player.getCurrentPosition()));
+        if (adPlaybackState != null && player != null && advertState != null) {
```

Comment:
The guards are essentially based on the output. Here, because we are stopping, we need to store the position (player), notify the player (AdPlaybackState) and reset the advert state.